### ISSUE TITLE
bug(reranker): HuggingFace reranker uses min-max normalization, collapsing single/tied results to 0.0

### ISSUE THREAD

**Issue by immuhammadfurqan** [mem0ai/mem0#5714]
### Component

Core / Python SDK (reranker)

### Description

`HuggingFaceReranker` normalizes cross-encoder scores with **min-max scaling** across the retrieved candidate set:

```python
# mem0/reranker/huggingface_reranker.py
scores = (scores - scores.min()) / (scores.max() - scores.min() + 1e-8)
```

This has two problems:

1. **Scores are set-relative.** The lowest-ranked document is *always* forced to `0.0`, even when it is highly relevant, and the same document gets a different `rerank_score` depending on what else was retrieved alongside it. Any downstream threshold on `rerank_score` is therefore unreliable.
2. **Single or tied results collapse to ~0.0.** When every score is equal — a single returned document, or several tied candidates — `max == min`, so every normalized score becomes `(0) / (1e-8) ≈ 0.0`. A reranker reporting its only result as `0.0` relevant is clearly wrong.

Cross-encoder rerankers (e.g. the default `BAAI/bge-reranker-base`) emit **unbounded logits**. The documented way to map them to an interpretable `[0, 1]` relevance score is a **per-document sigmoid**, which — per the sentence-transformers docs — does not change ranking order. mem0's own `SentenceTransformerReranker` does not min-max either.

### Steps to Reproduce

1. Configure a reranker with `provider: "huggingface"` (default `normalize=True`).
2. Run a search that returns a single memory, or several memories the model scores equally.
3. Inspect `rerank_score` on the results.

### Expected Behavior

Each document is scored on its own merit in `[0, 1]` (sigmoid of the logit); a single relevant result scores well above `0.5`.

### Actual Behavior

The single / tied results all receive `rerank_score = 0.0`, and in the multi-document case the least-relevant doc is always `0.0` regardless of its actual relevance.

### References

- BAAI/bge-reranker model cards: raw logits mapped to [0,1] via sigmoid.
- sentence-transformers docs: sigmoid yields [0,1] scores and does not affect ranking.

**Comment by kartik-mem0** [comment]
## Triage Notes

**What we've established:**
- `HuggingFaceReranker` in `mem0/reranker/huggingface_reranker.py` normalizes cross-encoder logits with `(scores - scores.min()) / (scores.max() - scores.min() + 1e-8)` — a set-relative transform that forces the lowest-scored document to exactly `0.0` regardless of its absolute relevance, and collapses single or tied results to `~0.0`.
- The sentence-transformers documentation recommends per-document sigmoid for mapping cross-encoder logits to `[0, 1]`, which gives each document an independent relevance estimate and does not change ranking order.
- `SentenceTransformerReranker` in the same package does not use min-max, so this is inconsistent within the reranker module.
- The ranking order is unaffected — this is a `rerank_score` correctness issue, not a result-ordering bug — but any downstream threshold on `rerank_score` is unreliable, and single-result cases look completely irrelevant at `0.0`.

**Reproduction:** Confirmed (code tracing; the single-document collapse is a deterministic consequence of `max == min` → all scores = `0 / 1e-8 ≈ 0.0`)
**Root cause:** `mem0/reranker/huggingface_reranker.py` normalizes with min-max rather than per-document sigmoid; fix is replacing the normalization block with `torch.sigmoid(scores)`.

**Next step:** PR #5715 addresses this — review feedback will be posted there.
